src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -1,124 +1,241 @@
-﻿using System.Collections.Generic;
-using System.Linq;
+﻿using System;
+using System.Collections.Generic;
 using Godot;
 
 public class ChemoreceptorUpgradeGUI : VBoxContainer, IOrganelleUpgrader
 {
     [Export]
-    public NodePath? CompoundsPath;
+    public NodePath? TargetTypesPath;
+
+    [Export]
+    public NodePath CompoundsPath = null!;
+
+    [Export]
+    public NodePath CompoundsLabelPath = null!;
+
+    [Export]
+    public NodePath SpeciesPath = null!;
+
+    [Export]
+    public NodePath SpeciesLabelPath = null!;
 
     [Export]
     public NodePath MaximumDistancePath = null!;
 
     [Export]
     public NodePath MinimumAmountPath = null!;
 
+    [Export]
+    public NodePath MinimumAmountLabelPath = null!;
+
     [Export]
     public NodePath ColourPath = null!;
 
 #pragma warning disable CA2213
+    private OptionButton targetTypes = null!;
     private OptionButton compounds = null!;
+    private Label compoundLabel = null!;
+    private OptionButton species = null!;
+    private Label speciesLabel = null!;
     private Slider maximumDistance = null!;
     private Slider minimumAmount = null!;
+    private Label minimumAmountLabel = null!;
     private TweakedColourPicker colour = null!;
 #pragma warning restore CA2213
 
-    private List<Compound>? shownChoices;
+    private List<Compound>? shownCompoundChoices;
+    private List<Species>? shownSpeciesChoices;
+
+    private enum TargetType
+    {
+        Compound,
+        Species,
+    }
 
     public override void _Ready()
     {
+        targetTypes = GetNode<OptionButton>(TargetTypesPath);
         compounds = GetNode<OptionButton>(CompoundsPath);
+        compoundLabel = GetNode<Label>(CompoundsLabelPath);
+        species = GetNode<OptionButton>(SpeciesPath);
+        speciesLabel = GetNode<Label>(SpeciesLabelPath);
         maximumDistance = GetNode<Slider>(MaximumDistancePath);
         minimumAmount = GetNode<Slider>(MinimumAmountPath);
+        minimumAmountLabel = GetNode<Label>(MinimumAmountLabelPath);
         colour = GetNode<TweakedColourPicker>(ColourPath);
 
         compounds.Clear();
+        species.Clear();
 
         maximumDistance.MinValue = Constants.CHEMORECEPTOR_RANGE_MIN;
         maximumDistance.MaxValue = Constants.CHEMORECEPTOR_RANGE_MAX;
-
         minimumAmount.MinValue = Constants.CHEMORECEPTOR_AMOUNT_MIN;
         minimumAmount.MaxValue = Constants.CHEMORECEPTOR_AMOUNT_MAX;
+
+        TypeChanged((int)TargetType.Compound);
     }
 
-    public void OnStartFor(OrganelleTemplate organelle)
+    public void OnStartFor(OrganelleTemplate organelle, GameProperties currentGame)
     {
-        shownChoices = SimulationParameters.Instance.GetCloudCompounds();
+        shownCompoundChoices = SimulationParameters.Instance.GetCloudCompounds();
+        shownCompoundChoices.Sort((x, y) => string.Compare(x.Name, y.Name, StringComparison.OrdinalIgnoreCase));
```

Comment:
Isn't there a "cleaner" sort overload something like `shownCompoundChoices.Sort(i => i.Name, StringComparer.OrdinalIgnoreCase)`? That avoids any potential bugs in manually performing the compare instead of just telling the standard sort algorithm to use a specific property.

Reply:
An alternative would be `shownCompoundChoices = shownCompoundChoices.OrderBy(x=>x.Name).ToList();`, but this is slower since it allocates a new list. With a quick search i couldn't find a way to shorten the `Comparison<T>`.

Reply:
Hang on, there's a bug here *anyway* as you are modifying the list that is owned by SimulationParameters. You should not do that. So the actual solution here is:

```
        shownCompoundChoices = SimulationParameters.Instance.GetCloudCompounds()
            .OrderBy(c => c.Name, StringComparer.OrdinalIgnoreCase).ToList();
```
(I had forgotten that the nice syntax is only supported by the LINQ sort and not the inbuilt list sort)

I'll make a separate change to make the simulation parameters return a unmodifiable list here.

Reply:
This change is now fine. And for reference here's the other PR I created: https://github.com/Revolutionary-Games/Thrive/pull/4454